### ISSUE TITLE
GLSL shader file import gets confused

### ISSUE THREAD

**Issue by mlepage** [godotengine/godot#72718]
### Godot version

4.0 beta 17

### System information

Windows 11, Intel integrated graphics

### Issue description

GLSL shader text files can confuse Godot 4 import pipeline, such that Godot still thinks they are shader files and can reimport them (recompiling etc.) but if there's an error, it cannot show the error. This can be "fixed" by copy/renaming the file, but that fix seems temporary, the fix is lost after time. This can be reproduced in beta 16 and 17 at least, detailed steps below.

### Steps to reproduce

Filesystem dock, new textfile, give it a name "foo.glsl", but "foo.glsl.txt" is created.

Right click new text file "foo.glsl.txt", rename to "foo.glsl".

Double click it, import opens and shows it is a GLSL Shader File.

Open it in the Godot 4 script editor. Enter some simple shader text that is correct:

```
#[compute]
#version 450
const float foo = 1.0;
void main() {}
```

Highlight "foo.glsl" in the filesystem dock and click reimport in the import tab. It should import without errors.

Now, type some garbage into "foo.glsl" so it should no longer compile when reimported.

Reimport. An error dialog should appear, saying there was an error importing the GLSL shader file, and to open the file via the filesystem dock to see the import error.

Double click on "foo.glsl" in the filesystem dock. This should open the file and show a ShaderFile tab with the compile output and error, _but it does nothing_. This can be reproduced in betas 16 and 17 (at least).

Godot still thinks the resource is type GLSL Shader File in the import tab UI, and in the "foo.glsl.import" file (which says type is RDShaderFile), but at the same time Godot is confused somewhere when importing the file. This can be shown by the next steps.

Go into local filesystem and copy "foo.glsl" to a new file "bar.glsl". It should still have the text that does not compile.

Godot will probably already have tried to import it and shown an error dialog. Just to confirm, highlight the new file "bar.glsl" in the filesystem dock and reimport it in the import tab, it should again state there is an error during import (but not what that error is).

Now, double click "bar.glsl" in the filesystem dock. It should open the ShaderFile tab and show compile output including details of the shader compilation error. _This is the step that did not work before but works now._

**To summarize:** we now have two identical files, "foo.glsl" and "bar.glsl", which are both same import type, and both do not import, but _only the latter shows the actual error when you double click it in the filesystem dock_.

**One more thing:** the fix for "bar.glsl" seems to eventually revert again to acting like "foo.glsl" and not being able to show the errors. I say this because I had done copy/rename fix days ago while working on my shader in beta 16, and after several sessions of opening/closing Godot, and upgrading to beta 17, I notice today that my shader has exhibited the import confusion problem again, and won't open to show details of compilation errors.

### Minimal reproduction project

N/A

**Comment by clayjohn** [comment]
Duplicate of https://github.com/godotengine/godot/issues/72281

The problem here is that you are trying to use the Godot text editor as a GLSL editor which it is not designed to do. If you edit the GLSL file from an external editor, Godot will automatically detect changes and reimport appropriately.

**Comment by mlepage** [comment]
Hold on, this isn't exactly the same issue. The other issue is that the file, when edited in Godot's editor, does not automatically reimport (as just about everything else does, so that's suprising). The file can still be (re)imported manually.

This issue, is that notwithstanding the above, you cannot see shader compilation errors in Godot (as Godot promises in the import failed dialog). How is that not a bug?

**Comment by clayjohn** [comment]
> Hold on, this isn't exactly the same issue. The other issue is that the file, when edited in Godot's editor, does not automatically reimport (as just about everything else does, so that's suprising). The file can still be (re)imported manually.
> 
> This issue, is that notwithstanding the above, you cannot see shader compilation errors in Godot (as Godot promises in the import failed dialog). How is that not a bug?

It's two parts of the same bug. For our sake, it is much better to add more information to the original bug report than it is to open a brand new bug report for what amounts to the same issue. That way when a contributor goes to fix https://github.com/godotengine/godot/issues/72281, they can get the whole picture of the bug (i.e. its not just impacting automatic reloading, it is also breaking compilation errors).

For us, the ideal workflow would be that users search through open issues and comment additional information on existing bug reports rather than opening duplicates to provide more information.
